src/main/java/com/google/sps/servlets/Recipe.java

```diff
@@ -134,4 +134,17 @@ public void setSteps(ArrayList<String> givenSteps) {
   public void setIngredients(ArrayList<String> givenIngredients) {
     ingredients = givenIngredients;
   }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == this) {
+      return true;
+    }
+    if (obj == null || obj.getClass() != this.getClass()) {
+      return false;
+    }
```

Comment:
Why do you need these two?

Reply:
I used this when I implemented the search by title & ingredients using two lists, which is more memory consuming than this implementation.